apisix/plugins/log-rotate.lua

```diff
@@ -117,6 +121,22 @@ local function rotate_file(date_str, file_type)
     local ok, msg = os.rename(file_path_org, file_path)
     core.log.info("move file from ", file_path_org, " to ", file_path,
                   " res:", ok, " msg:", msg)
+
+    if ok and enable_compression then
+        local compression_filename = new_filename .. COMPRESSION_FILE_SUFFIX
+        local cmd = string.format("cd %s && tar -zcf %s %s",
+            log_dir, compression_filename, new_filename)
+        core.log.info(cmd)
+        ok = os.execute(cmd)
```

Comment:
Better to use lua-resty-shell

Reply:
yep, use lua-resty-shell need add a new package `resty.shell` to this project. currently use the common `os`.